### PR TITLE
Implement string length and substring

### PR DESCRIPTION
**Pull request by ltratt**

### CODE REVIEW

File: src/lib/vm/objects/string_.rs [hunk 1]
```diff
@@ -128,6 +128,25 @@ impl String_ {
         Ok(String_::new_str(vm, new))
     }
 
+    pub fn substring(&self, vm: &mut VM, start: usize, end: usize) -> Result<Val, Box<VMError>> {
+        if start == 0 || end == 0 {
```
Comment (ptersilie) [inline src/lib/vm/objects/string_.rs:132]: What about `start < 0`?

Comment (ltratt) [reply src/lib/vm/objects/string_.rs:132]: They're `usize` so they can't be `< 0`.

Comment (ptersilie) [reply src/lib/vm/objects/string_.rs:132]: Oh I missed that.

**Pull request merged**